.github/workflows/test.yml

```
@@ -0,0 +1,91 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  render:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - run: python -m pip install cookiecutter
+      - run: python -m cookiecutter . --no-input
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: rendered-project
+          path: wagtail-llama-save
+
+  lint:
+    needs: render
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - uses: actions/download-artifact@v3
+        # Restore the rendered areifact to a directory outside the repo checkout.
+        # Even if we invoke pre-commit from in a subdirectory
+        # it tries to lint the whole repo including the raw templates.
+        with:
+          name: rendered-project
+          path: /tmp/wagtail-llama-save
+
+      - name: Run Lint Checks
+        working-directory: /tmp/wagtail-llama-save
+        # In order to get pre-commit to see any files,
+        # we have to initialize a git repo
+        # and at least stage the files we want to check.
+        # Otherwise pre-commit will stash everything and
+        # tell us there are no files to check.
+        # refs
+        # https://github.com/wagtail/cookiecutter-wagtail-package/pull/45#discussion_r1064195179
+        # https://github.com/pre-commit/pre-commit/issues/848
+        run: |
+          git init
+          git add .
+          python -m pip install pre-commit
+          pre-commit run --all-files --show-diff-on-failure --color=always
+
+  test:
+    needs: render
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: rendered-project
+          path: wagtail-llama-save
+
+      - working-directory: wagtail-llama-save
+        run: python -m pip install -e .[testing]
+      - working-directory: wagtail-llama-save
+        run: python testmanage.py test
+
+  build-js:
+    needs: render
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: rendered-project
+          path: wagtail-llama-save
+
+      - working-directory: wagtail-llama-save
+        run: npm install
+      - working-directory: wagtail-llama-save
+        run: npm run build
```

question: could this benefit from `npm ci --no-optional --no-audit --progress=false` ?

also: 
```suggestion
      - working-directory: wagtail-llama-save
        run: | 
          npm ci --no-optional --no-audit --progress=false
          npm run build
```

?

---

`npm ci` requires a lockfile and we don't have one (correctly, I think) so it has to be `npm install` but otherwise I've addressed your feedback in ffb9317 and 734f2ae